CONTRIBUTING.md

```diff
@@ -22,7 +23,11 @@ Gonum discussion takes place on the [gonum-dev google group](https://groups.goog
 
 ### Code of Conduct
 
-In general follow the [Go code of conduct](https://golang.org/conduct).  If you are a contributor, please be welcoming to new contributors.  [Here](http://sarah.thesharps.us/2014/09/01/the-gentle-art-of-patch-review/) is a good guide.  If you are a new contributor, thank you!
+We expect the same values and behavior as described in [Go code of conduct](https://golang.org/conduct).
+
+### Project Scope
+
+The purpose of the Gonum project is to provide general purpose numerical computing and data science libraries for the Go programming language.  The libraries should aim to provide *primatives* not *solutions*.  Code should be implemented first in pure Go, and the use of assembly, C, Fortran, or other languages should be optional, and justified with performance considerations.
```

Comment:
How about "The libraries should aim to provide building blocks for disciplinary work and advanced algorithms"

I don't like "C ... should be optional". Maybe "Code should be implemented in pure go. Calls to other languages, such as assembly, C, or Fortran, may be included with justified performance considerations, but should always be opt-in for users.